docs/design/security.md

```diff
@@ -1,17 +1,106 @@
+
+
+1. The API should authenticate and authorize user actions [authn and authz](https://github.com/GoogleCloudPlatform/kubernetes/blob/master/docs/design/access.md)
+2. All infrastructure components (kubelets, kube-proxies, controllers, scheduler) should have an infrastructure user that they can authenticate with and be authorized to perform only the functions they require against the API.
+3. Most infrastructure components should use the API as a way of exchanging data and changing the system, and only the API should have access to the underlying data store (etcd)
+4. When containers run on the cluster and need to talk to other containers or the API server, they should be identified and authorized clearly as an autonomous process via a [service account](https://github.com/GoogleCloudPlatform/kubernetes/pull/2297)
+   a.  If the user who started a long-lived process is removed from access to the cluster, the process should be able to continue without interruption
+   b.  If the user who started processes are removed from the cluster, administrators may wish to terminate their processes in bulk
+   c.  When containers run with a service account, the user that created / triggered the service account behavior must be associated to the container's action
+5. When container processes runs on the cluster, they should run in a [security context](https://github.com/GoogleCloudPlatform/kubernetes/pull/3910) that isolates those processes via Linux user security, user namespaces, and permissions.
+   a.  Administrators should be able to configure the cluster to automatically confine all container processes as a non-root, randomly assigned UID
+   b.  Administrators should be able to ensure that container processes within the same namespace are all assigned the same unix user UID
+   c.  Administrators should be able to limit which developers and project administrators have access to higher privilege actions
+   d.  Project administrators should be able to run pods within a namespace under different security contexts, and developers must be able to specify which of the available security contexts they may use
+   e.  Developers should be able to run their own images or images from the community and expect those images to run correctly
+   f.  Developers may need to ensure their images work within higher security requirements specified by administrators
+   g.  When available, Linux kernel user namespaces can be used to ensure b. and d. are met.
+   h.  When application developers want to share filesytem data via distributed filesystems, the Unix user ids on those filesystems must be consistent across different container processes
+6. Developers should be able to define [secrets](https://github.com/GoogleCloudPlatform/kubernetes/pull/2297) that are automatically added to the containers when pods are run
+   a.  Secrets are files injected into the container whose values should not be displayed within a pod. Examples:
```

Comment:
clarify if "should not be displayed within a pod" means "should not be displayed by GET /pods" or "should not be visible to other containers in the same pod."  I think you mean the former.